### ISSUE TITLE
Memory Leak when calling FirebaseApp.configure

### ISSUE THREAD

**Issue by edys00** [firebase/firebase-ios-sdk#11071]
### Description

When I was investigating memory leak in our App, after eliminating some of potential codes I found that one of the memory leak issues is actually caused by Firebase iOS SDK and it happens after calling `FirebaseApp.configure`. I attached the screenshot from memory graph debugger, unfortunately there is no useful information from the backtrace but can confirm all the APMP thing is from Firebase SDK.
<img width="307" alt="Screen Shot 2023-04-04 at 11 30 01" src="https://user-images.githubusercontent.com/40975424/229687570-f13f6999-e82a-46d3-9c98-59487c9f80fc.png">
<img width="1511" alt="Screen Shot 2023-04-04 at 11 29 54" src="https://user-images.githubusercontent.com/40975424/229687584-57a93c5e-ab41-45ac-83d7-ff968b438995.png">

### Reproducing the issue

Call `FirebaseApp.configure` on App Launch and turn on `Debug Memory Graph` after the app finish launching.

### Firebase SDK Version

10.2.0

### Xcode Version

14.1

### Installation Method

Zip

### Firebase Product(s)

AB Testing, Analytics, Crashlytics, DynamicLinks, In-App Messaging, Installations, Messaging, Performance, Remote Config

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
APMP prefixed symbols are from FirebaseAnalytics

**Comment by skuske** [comment]
Possibly identical to https://github.com/firebase/firebase-ios-sdk/issues/11027

**Comment by paulb777** [comment]
@skuske Instruments is showing the leak in a FirebaseAnalytics module in this report versus a FirebaseSessions module in #11027 - so these are almost certainly different leaks.

**Comment by strahinja-mem** [comment]
I'm getting this leak as well. Any news on the fix?

**Comment by tsunghung** [comment]
Thank you for reporting this issue. We have confirmed the issue and are working on a fix. However, the fix requires a number of changes, so it may not be available soon.

The good news is that the memory leak appears to be a fixed amount, i.e. it won't increase over time, so users shouldn't see much of an impact.

**Comment by edys00** [comment]
thanks for the update @tsunghung.

looking forward to the fix being rolled out when ready.

**Comment by Anton-Ivchenko-Axon** [comment]
any news?

**Comment by tsunghung** [comment]
Sorry about that. Please see [the comment](https://github.com/firebase/firebase-ios-sdk/issues/11071#issuecomment-1510734813).

_We have confirmed the issue and are working on a fix. However, the fix requires a number of changes, so it may not be available soon. The good news is that the memory leak appears to be a fixed amount, i.e. it won't increase over time, so users shouldn't see much of an impact._

**Comment by Invader1999** [comment]
Any updates on the fix even iam facing the issue

**Comment by jotai-coder** [comment]
We are also seeing this.

**Comment by Ilyeesz** [comment]
We are facing the same issue for the past six months, any updates ?

**Comment by tsunghung** [comment]
Sorry for the long wait. The fix will be available in the next release in early March.

**Comment by andrewheard** [comment]
Sorry again for the long wait. The fix is now available in the [Firebase 10.22.0 release](https://github.com/firebase/firebase-ios-sdk/releases/tag/10.22.0).

**Comment by edys00** [comment]
noted, thanks for the assistance @paulb777 @tsunghung @andrewheard !
